File: vol1/ch7/LeetCode230.cpp
class Solution {
    int cnt;
    int ans;
public:
    int kthSmallest(TreeNode* root,int k) {
        cnt=0;
        inorder(root,k);
        return ans;

    }
    bool inorder(TreeNode*root,int k) {
        if(root==NULL) return false;
        if(inorder(root->left,k)) return true;
        cnt++;
        if(cnt==k) {
            ans=root->val;
            return true;
        }
        return inorder(root->right,k);
    }
};

File: vol1/ch7/LeetCode230.py
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.cnt=0
        self.inorder(root,k)
        return self.ans

    def inorder(self,root,k):
        if root==None: return False
        if self.inorder(root.left,k): return True
        self.cnt+=1
        if self.cnt==k:
            self.ans=root.val
            return True
        return self.inorder(root.right,k)
